### PR TITLE
Write quarkus-core this extension depends on in quarkus-extension.yaml

### PR DESCRIPTION
**Pull request by gastaldi**
The quarkus-extension.yaml now has the `quarkus-core` version used when the extension is built. This is useful for tooling when dealing with third-party extensions to avoid traversing the dependency tree

### CODE REVIEW

File: independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java [hunk 1]
```diff
@@ -274,16 +283,50 @@ private void transformLegacyToNew(final Path output, ObjectNode extObject, Objec
...
+        }
         extObject.set("metadata", metadata);
+    }
 
-        //   updateSourceFiles(output, extObject, mapper);
+    private String resolveQuarkusCoreVersion() throws MojoExecutionException {
+        MavenArtifactResolver mvn = getMavenResolver();
+        DefaultArtifact artifact = new DefaultArtifact(project.getGroupId(),
+                project.getArtifactId(),
+                project.getPackaging(),
```
Comment (aloubyansky) [inline independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java:298]: 'packaging' is not the same as the type. `projectArtifact.getArtifactHandler().getExtension()` will be more appropriate.

File: independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java [hunk 2]
```diff
@@ -274,16 +283,50 @@ private void transformLegacyToNew(final Path output, ObjectNode extObject, Objec
...
+            public boolean visitEnter(DependencyNode node) {
+                return version.get() == null;
+            }
 
+            @Override
+            public boolean visitLeave(DependencyNode node) {
+                final Dependency dep = node.getDependency();
+                if (dep != null
+                        && dep.getArtifact().getArtifactId().equals("quarkus-core")
+                        && dep.getArtifact().getGroupId().equals("io.quarkus")) {
```
Comment (aloubyansky) [inline independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java:318]: There are extensions that don't depend on `quarkus-core` and it's not necessarily a mistake. However every deployment artifact has `quarkus-core-deployment` among its deps. So, that's what you should be looking for instead.

**Review by aloubyansky — CHANGES_REQUESTED** [review]

File: independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java (hunk 2 above)
Comment (gastaldi) [reply independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java:318]: The problem is that the `quarkus-extension.yaml` is in the runtime module, and the deployment module is a dependency of of this module

Comment (aloubyansky) [reply independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java:318]: Is there a reason to keep it in the runtime artifact?

Comment (gastaldi) [reply independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java:318]: Dunno, it has always been like this AFAIK.

**Comment by gastaldi** [comment]
Closing for now

**Pull request closed without merging**